src/main/kotlin/com/hedvig/underwriter/service/quotesSignDataStrategies/SignStrategyService.kt

```diff
@@ -113,6 +110,12 @@ class SignStrategyService(
             this.any { it.data is DanishTravelData }
 
     override fun getSignMethod(quotes: List<Quote>): SignMethod {
+        val isSelfChangeQuotes = quotes.any { it.initiatedFrom == QuoteInitiatedFrom.SELF_CHANGE }
```

Comment:
Insure how to do this. Guess we need to validate the opposite as well 🤔 
```suggestion
        val isSelfChangeQuotes = quotes.all { it.initiatedFrom == QuoteInitiatedFrom.SELF_CHANGE }
```

Reply:
Or maybe use `any` and break further down if it was a mixed bag

Reply:
Well, I think the root cause is the model being a "list of quotes" and not a single quote. They need to have certain properties perfectly aligned.

The mega-defensive solution would be to crash if the `initiatedFrom` value is not the same.

But my half-defensive model here is to say that "if one of them is SELF_CHANGE", then `approveOnly` is the sign method. Then in the sign method itself, we fail if they are not all `SELF_CHANGE`.

Reply:
I don't find that any of them are obviously better honestly

Reply:
Yea signing is weird. I say go with this